build.gradle

```diff
@@ -101,8 +97,40 @@ apply from: 'deploy.gradle'
 apply from: 'tasks.gradle'
 apply from: 'jmh.gradle'
 
+project(':iceberg-relocate-external-dependencies') {
```

Comment:
Since this is minimized, we can't use it for Jackson. What about separate modules for both? So this would be `iceberg-shaded-guava` and the other would be `iceberg-shaded-jackson`.

Reply:
Is it worth the effort to minimize? If we just put all of Guava (and Jackson in the future) then this could just produce one relocated jar file. It also means one doesn't have to have the extra step of updating the GuavaClasses file every time you rely on a new Guava class. 

Alternatively we could create a subproject per "problematic" external library I'm just concerned we'd end up with quite a few of them but on the other hand it would allow you to pick and choose them in other projects based on which ones that project actually uses.

You guys choose, if you want to keep it minimized and one per library then I can rename this accordingly to be just guava for now.

Reply:
I'm ok with not being minimized.

Reply:
I prefer the shaded approach because so many projects have to do this with Guava. Avro and Parquet already do this, so we're already asking people to include 3 copies of Guava.

I also don't think we are going to have to do this very often. Once this is set up we shouldn't be changing it much. And we should avoid adding new libraries that require it. If we end up with just Jackson and Guava, I think it will be fine.